packages/babel-parser/src/parser/comments.ts

```diff
@@ -7,32 +7,20 @@ import type { Undone } from "./node";
 
 /**
  * A whitespace token containing comments
- * @typedef CommentWhitespace
- * @type {object}
- * @property {number} start - the start of the whitespace token.
- * @property {number} end - the end of the whitespace token.
- * @property {Array<Comment>} comments - the containing comments
- * @property {Node | null} leadingNode - the immediately preceding AST node of the whitespace token
- * @property {Node | null} trailingNode - the immediately following AST node of the whitespace token
- * @property {Node | null} containingNode - the innermost AST node containing the whitespace
- *                                         with minimal size (|end - start|)
  */
 export type CommentWhitespace = {
-  start: number;
-  end: number;
-  comments: Array<Comment>;
-  leadingNode: Node | null;
-  trailingNode: Node | null;
-  containingNode: Node | null;
+  start: number; // the start of the whitespace token.
```

Comment:
Nit, but if we use jsdoc-like comments editors will show the as the description in autocompletion:
```js
  /**
   * The start of the whitespace token.
   */
  start: number;
```

even if probably comments for start/end/comments are not needed.

Reply:
It should work when overing over start:

![image](https://user-images.githubusercontent.com/7000710/207644862-b2e0ce9b-7d46-474e-bd4d-f9be6ef4875c.png)